packages/@react-types/table/src/index.d.ts

```diff
@@ -10,72 +10,109 @@
   renderEmptyState?: () => JSX.Element
 }
 
 export interface TableHeaderProps<T> {
+  /** A list of Table columns. */
   columns?: T[],
+  /** A list of `Column(s)` or a function. If the latter, a list of columns must be provided using the `columns` prop. */
   children: ColumnElement<T> | ColumnElement<T>[] | ColumnRenderer<T>
 }
 
 type ColumnElement<T> = ReactElement<ColumnProps<T>>;
 type ColumnRenderer<T> = (item: T) => ColumnElement<T>;
 export interface ColumnProps<T> {
+  /** Rendered contents of the column if `children` contains child columns. */
   title?: ReactNode,
+  /** Static child columns or content to render as the column header. */
   children: ReactNode | ColumnElement<T> | ColumnElement<T>[],
+  /** A list of child columns used when dynamically rendering nested child columns. */
   childColumns?: T[],
-  'aria-label'?: string,
+  /** The width of the column. */
   width?: number | string,
+  /** The minimum width of the column. */
   minWidth?: number | string,
+  /** The maximum width of the column. */
   maxWidth?: number | string,
+  /** The default width of the column. */
   defaultWidth?: number | string
 }
 
 // TODO: how to support these in CollectionBuilder...
 export interface SpectrumColumnProps<T> extends ColumnProps<T> {
+  /**
+   * The alignment of the column's contents relative to its allotted width.
+   * @default 'start'
+   */
   align?: 'start' | 'center' | 'end',
-  allowsResizing?: boolean,
-  allowsReordering?: boolean,
+  /** Whether the column allows sorting. */
   allowsSorting?: boolean,
-  isSticky?: boolean, // shouldStick??
+  // /** Whether the column should stick to the viewport when scrolling. */
+  // isSticky?: boolean, // shouldStick?? Not implemented yet?
+  /** Whether a column is a row header and should be announced by assistive technology during row navigation. */
   isRowHeader?: boolean,
+  /** Whether the column should render a divider between it and the next column. */
   showDivider?: boolean,
+  /**
+   * Whether the column should hide its header text. A tooltip with the column's header text
+   * will be displayed when the column header is focused instead. Note that this prop is specifically for columns
+   * that contain ActionButtons in place of text content.
+   */
   hideHeader?: boolean
 }
 
 export interface TableBodyProps<T> extends Omit<AsyncLoadable, 'isLoading'> {
+  /** The contents of the table body. Supports static items or a function for dynamic rendering. */
   children: CollectionChildren<T>,
+  /** A list of row objects in the table body used when dynamically rendering rows. */
   items?: Iterable<T>,
   /** The current loading state of the TableView. */
   loadingState?: LoadingState
 }
 
 export interface RowProps<T> {
   // treeble case?
+  /** A list of child item objects used when dynamically rendering row children. */
   childItems?: Iterable<T>,
+  /** Whether this row has children, even if not loaded yet. */
   hasChildItems?: boolean,
```

Comment:
IMO we should remove these props for now since we don't support nested rows yet. You might need to comment out some implementation in Row.ts as well.